### ISSUE TITLE
master version of kubernetes-mixin jsonnet lib cause error due to the latest update

### ISSUE THREAD

**Issue by ArvinDevel** [coreos/prometheus-operator#1496]
**What did you do?**
extend prometheus rule

**What did you expect to see?**
run well

**What did you see instead? Under which circumstances?**
manifestation error

**Environment**
Linux server, mac client

* Kubernetes version information:

Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-13T22:27:55Z", GoVersion:"go1.9.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.6", GitCommit:"6260bb08c46c31eea6cb538b34a9ceb3e406689c", GitTreeState:"clean", BuildDate:"2017-12-21T06:23:29Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster kind: 

kubeadm

* Manifests:

```
local kp = (import 'kube-prometheus/kube-prometheus.libsonnet') + {
  _config+:: {
    namespace: 'monitoring',
  },
  prometheusAlerts+:: {
    groups+: [
      # TODO: use import
      #'self-defined-alert.rules': (import 'self-defined-alert.rules.yaml'),
      {
        name: 'self-defined-alert',
        rules: [
          {
            # Alert for any pod whose CpuUsage is above 85 for 1h.
            alert: 'PodCpuUsageHigh',
            expr: 'avg (rate(container_cpu_usage_seconds_total[10m])) by (pod_name) > 0.85',
            'for': '1h',
            labels: {
              severity: 'warning',
            },
            annotations: {
              summary: 'High Cpu Usage above 85% of {{ $labels.pod_name }} for 1h',
              description: 'This is a cpu usage alert for pod',
            },
          },
        ],
      },
    ],
  },
};

# just generate prometheus related manifests
{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) }
```

[tutorial](https://github.com/coreos/prometheus-operator/blob/master/contrib/kube-prometheus/docs/developing-prometheus-rules-and-grafana-dashboards.md)

* Prometheus Operator Logs:

```
insert Prometheus Operator logs relevant to the issue here
```

* Related PR in kubernetes-mixin
[cause pr](https://github.com/kubernetes-monitoring/kubernetes-mixin/commit/3c4224883f6916e5e21a8555d42b65277f2a0a7f)

I wonder should we use one solid version of the jsonnet dependency lib to avoid this?

**Comment by brancz** [comment]
You're right, I'm currently fixing this.

> I wonder should we use one solid version of the jsonnet dependency lib to avoid this?

Yes, unfortunately there is a bug in jsonnet-bundler right now that needs to be fixed for this to work correctly.

**Comment by brancz** [comment]
The concrete issue here was fixed by #1497 and the respective upstream PR. The remaining jsonnet-bundler issue is tracked on that repository.
